### PR TITLE
core changes to allow for workflow transaction deletion

### PR DESCRIPTION
**Pull request by aarongundel**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

This change depends on PR #7595 and is branched off it.
Core changes to allow for workflow transaction deletion. 
* Adds transaction ID to tile viewmodel.  
* Adds workflow ID to component based step.
* Adds the transaction reversal endpoint to quitWorkflow in workflow.js

### Issues Solved

#7587 - makes it possible.  But individual workflows will need to be modified to pass the transaction ID/workflow ID over.

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

File: arches/app/media/js/viewmodels/workflow.js [hunk 1]
```diff
@@ -301,6 +301,14 @@ define([
             });
         };
 
+        this.reverseWorkflowTransactions = function() {
+            $.ajax({
```
Comment (chrabyrd) [inline arches/app/media/js/viewmodels/workflow.js:305]: Overall everything looks good, just one nit:

we're using a `GET` here to roll back the DB, yeah? Can this be a POST instead?

**Review by chrabyrd — CHANGES_REQUESTED** [review]

File: arches/app/media/js/viewmodels/workflow.js (hunk 1 above)
Comment (aarongundel) [reply arches/app/media/js/viewmodels/workflow.js:305]: Good idea - done.

**Comment by aarongundel** [comment]
Also fixed merge conflict

File: arches/app/media/js/viewmodels/workflow.js [hunk 2]
```diff
@@ -301,6 +301,14 @@ define([
...
+                type: "POST",
+                url: arches.urls.transaction_reverse(self.id()),
+                context: self,
```
Comment (chrabyrd) [inline arches/app/media/js/viewmodels/workflow.js:308]: Do we need to pass the entire workflow as context in this call?

**Review by chrabyrd — CHANGES_REQUESTED** [review]

File: arches/app/media/js/viewmodels/workflow.js (hunk 2 above)
Comment (aarongundel) [reply arches/app/media/js/viewmodels/workflow.js:308]: I don't think so since there's no callback on this POST anyway.  Will remove.

**Review by chrabyrd — APPROVED** [review]
works as expected

**Pull request merged**